### PR TITLE
Add a Mixer integration test for testing K8s integration.

### PR DESCRIPTION
**Pull request by ozevren**

### CODE REVIEW

File: pkg/test/framework/components/istio/kube.go [hunk 1]
```diff
@@ -116,17 +116,20 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 		return nil, err
 	}
 
-	// Wait for Galley & the validation webhook to come online before applying Istio configurations.
-	if _, _, err = env.WaitUntilServiceEndpointsAreReady(cfg.SystemNamespace, "istio-galley"); err != nil {
-		err = fmt.Errorf("error waiting %s/istio-galley service endpoints: %v", cfg.SystemNamespace, err)
-		scopes.CI.Info(err.Error())
-		return nil, err
-	}
+	if !cfg.DoNotWaitForValidationWebhook {
```
Comment (kyessenov) [inline pkg/test/framework/components/istio/kube.go]: nit: double negation, suggest structure as a block with an early exit.

File: tests/integration/mixer/k8s/k8s_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	// Test to ensure that deployment in K8s environment completes. We can extend this suite in the future
+	// to include more Mixer/K8s integration tests. In that case, it may make sense to move the deployment
+	// logic into TestMain.
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+		cfg := istio.DefaultConfigOrFail(ctx, ctx)
+		cfg.Values["global.useMCP"] = "false"
+		cfg.Values["galley.enabled"] = "false"
+		cfg.DoNotWaitForValidationWebhook = true
+
+		_, err := istio.Deploy(ctx, &cfg)
```
Comment (kyessenov) [inline tests/integration/mixer/k8s/k8s_test.go]: Does this check that crashloops don't happen? What concerns me is that even in a regular install with Galley, Mixer crash loops in the beginning. So how do you tell a difference between intermittent crashes and hard crash without Galley?

Comment (kyessenov) [reply tests/integration/mixer/k8s/k8s_test.go]: IMHO we should run a basic Mixer-dependent functionality test to be sure it's actually working.

File: pkg/test/framework/components/istio/kube.go (hunk 1 above)
Comment (ozevren) [reply pkg/test/framework/components/istio/kube.go]: Renamed to Skip. This needs to be false by default.

File: tests/integration/mixer/k8s/k8s_test.go (hunk 2 above)
Comment (ozevren) [reply tests/integration/mixer/k8s/k8s_test.go]: It doesn't check for crashloops. It does ensure that all components come up (liveness & readiness) eventually.

Comment (ozevren) [reply tests/integration/mixer/k8s/k8s_test.go]: Sounds good, but I didn't want to copy paste some arbitrary test here, and I don't have the bandwidth to come up with a good factoring or a representative test. If you have a cheap recommendation, I am happy to incorporate.

Comment (kyessenov) [reply tests/integration/mixer/k8s/k8s_test.go]: There is a delay in the liveness/readiness probe signal showing up. I think it's on the order of 30s before k8s realizes it's dead. Maybe insert a 5 minutes delay and check for readiness after that?

Comment (ozevren) [reply tests/integration/mixer/k8s/k8s_test.go]: Nice catch. I added additional delay/checks. PTAL.

File: tests/integration/mixer/k8s/k8s_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		env := ctx.Environment().(*kube.Environment)
+
+		// Ensure that we can loop-through at least ten times while all pods in the ready.
+		for i := 0; i < 40; i++ {
+			if _, err := env.CheckPodsAreReady(env.NewPodFetch(cfg.IstioNamespace)); err != nil {
+				t.Fatalf("Error waiting for pods: %v", err)
```
Comment (kyessenov) [inline tests/integration/mixer/k8s/k8s_test.go]: Isn't this a bit harsh? Should we tolerate initial crash loops in k8s mode too at the beginning? I'm just worried that making it overly strict will lead to flakiness.

Comment (ozevren) [reply tests/integration/mixer/k8s/k8s_test.go]: Most of the initial crash loops are accounted in the original istio deployment pod readiness checks. The gateway readiness is generally the long pole.

Still added more resiliency.

**Comment by ozevren** [comment]
Can I get another lgtm?

**Pull request merged**